eventuals/compose.h

```diff
@@ -95,8 +96,49 @@ struct Composed final {
   template <typename Arg, typename K>
   auto k(K k) && {
     using Value = typename Left_::template ValueFrom<Arg>;
-    return std::move(left_)
-        .template k<Arg>(std::move(right_).template k<Value>(std::move(k)));
+
+    auto composed = [&]() {
+      return std::move(left_).template k<Arg>(
+          std::move(right_).template k<Value>(std::move(k)));
+    };
+
+#ifndef __APPLE__
```

Comment:
```suggestion
#if defined(_WIN32) || defined(__linux__)
```

Reply:
Agreed!

Reply:
For Windows there is a need to find other api for this stuff.

Reply:
As I got `pthread_get_stackaddr_np` and `pthread_get_stacksize_np` functions are only for macOS platform (see this [`doc`](https://opensource.apple.com/source/libpthread/libpthread-105.1.4/src/pthread.c.auto.html)).

Reply:
Ok, now I'm confused, if it's only available on MacOS, why do we have `#ifndef __APPLE__`?

Reply:
Also, we are using `__MACH__` in other files